### ISSUE TITLE
Change subheading on the homepage with paragraph

### ISSUE THREAD

**Issue by styamsingh** [jakewies/hugo-theme-codex#32]
Hey. First, I would say that this is really well built theme and I have been using it to build my personal website (it's not entirely done) and I have a suggestion that can be incorporated in the theme.

The subheading for the homepage is taken from `_index.md`'s `subheading` section but it's really limiting. First, you can't have a line break for typing multiple paragraphs. I think most people would want to type out more than a couple of lines (I wanted to type out a couple of paragraphs describing myself) and that's not possible with a subheading. Second, the secondary content should be taken from `_index.md`'s content section and rendered as `<p>` instead of `<h2>`.

I have tried to achieve the above by doing the following (I have little experience working with GO so the following code block as well as the subsequent code blocks would not be perfect and more of a work around to build the feature that I wanted):

```html
<!-- Delete this block of code from index.html -->
<h2>
    {{ .Params.subheading }}
</h2>

<!-- And replace the above code block with this code -->
<p>
    {{ .Content }}
</p>
```
I also defined some paragraph specific styles in `about.css` so as to make it look better.
```css
.splash p {
  font-size: 1.6em;
  font-weight: 400;
  line-height: 1.25;
  max-width: 30em;
}
```
And here's my _index.md
```yaml
---
heading: "Hi, I'm Satyam Singh"
handle: "stymsing"
---
I am a first-year engineering student at Indraprastha University (New Delhi), pursuing a bachelor's degree in Information Technology.

I ❤️ exploring various technologies relating to Computer Science and writing highly optimized and scalable code that contributes towards solving a real, bigger problem as well as shell scripts for automating repetitive tasks.

Besides programming, I like reading about space and playing computer games.
```
[This is how the homepage looks after implementing above changes](https://i.imgur.com/VOhnOAS.png)

### Bugs due to implementing the above feature
The above changes introduce a bug with how content of the homepage is viewed on smaller screens.

The homepage uses `main {height: 100%;}` to center the content vertically and it mostly does fine when there are only a couple of lines as the subheading (basically how the homepage was by default) but upon implementing the above changes and adding a couple of 3-4 line paragraphs, the content flows inside the hamburger menu. I have attached a few screenshots on how it looks on various small-screen devices:

[Homepage on iPhone 6/7/8](https://i.imgur.com/SeKWiSQ.png)
[Homepage on iPhone 5/SE](https://i.imgur.com/6Bsddi5.png)

I think what's happening is since the height of main tag is set to be 100%, the webpage is trying to fit whole of the content in viewable area (so that a scrollbar is not displayed/required) and hence in doing so it pushes the content upwards and inside of the hamburger menu. 

Now, I have tried to fix this issue by defining the height property only for large devices (>=800px as defined in the breakpoints) and changed the positioning of hamburger menu from `fixed` to `sticky`.

```css
.burger__container {
  position: sticky;
}

.
.
.

main {
  padding-top: 0;
  padding-bottom: 0;
}

@media screen and (min-width:800px) {
  main {
    height:100%;
  }
}
```
The above change pushes the content down and it doesn't flow inside the hamburger menu anymore and also introduces a scroll bar. I have attached a few screenshots of the homepage after the above change:

[Homepage on iPhone 6/7/8](https://i.imgur.com/lwq5pNZ.png)
[Homepage on iPhone 5/SE](https://i.imgur.com/TJBGKtH.png)

But now since I have restricted vertical centering only to larger devices (>=800px) this creates a problem with devices such as iPad since the content isn't vertically centered anymore (screenshot below). I haven't been able to solve this problem yet. Maybe defining flexbox separately for smaller screens would help.

[Homepage on iPad](https://i.imgur.com/K9wCquh.png)

**Comment by jakewies** [comment]
Hey @styamsingh ! Thanks for your interest in the theme. I'm glad it's been working out for you (for the most part).

You bring up an interesting point. The `index.html` page is quite limiting. But that is by design. My aim is to provide a constrained solution for those who want to get up and running quickly. 

I understand this isn't going to work for you, and you won't be the last to feel this way. That's why I think the best solution in this case would be to document how to override the `index.html` page completely if you so choose.

So how can we do that?

The first step would be to create a `layouts/index.html` file in your project directory. Hugo will use this over the one defined in the theme. From there you can copy pasta the content from `themes/hugo-theme-codex/layouts/index.html` into your own and make the structural adjustments you desire.

This solves the content issue but doesn't do anything to fix the css issues you ran into when making the change. You can take steps to alleviate this by adding your own stylesheet at `static/css/your-styles.css` and link to in your `/layouts/index.html` file:

```html
<!-- layouts/index.html -->

{{ define "stylesheets" }}
	<link rel="stylesheet" href="/css/about.css" /> <!-- DONT DELETE! -->
	<link rel="stylesheet" href="/css/your-styles.css" /> 
{{ end }}
```

It's important not to remove the theme's `about.css` file or you will lose all the things. 😅

Now you can go ahead and work on the styles that are unique to your site. I've implemented all of your changes to see what you're seeing and a quick fix I could give you, if you're willing to go this route, is to change the font-size of the `p` elements from `1.6em` to something smaller. This will bring your content down. Just a suggestion! 😅

Also I want to point out that there is a pattern going on here:

Anytime the theme is too constrained for you, you can override it by creating the folder / file in your own project. Hugo's lookup order will use your files before the theme (if they exist).

**Comment by jakewies** [comment]
Another option you might consider is to keep the homepage as is and add a separate page for more details on who you are. @chuxinh has done this [on her own site](https://www.chuxinhuang.com/) and it looks great.

But, I don't want to tell you how your website should look! If you want your "about me" to be detailed on the homepage, then by all means 😄

**Comment by styamsingh** [comment]
Thank you for taking the time to write a detailed answer @jakewies.

I didn't know about the overriding of files upon creating a new file of the same name in the root directory (and subdirectories) - definitely going to be useful while developing the website. I should have read the docs on Hugo's website before writing any code of my own 🤦  (well, in my defense, I skimmed through the documentation and found few Go code blocks which I didn't really understand, so I gave up and jumped right in 😜).

As far as the issue is concerned, I went ahead and reduced the `font-size` of the paragraph on smaller screens - as per you suggestion and it does the job (until I add some more content, but at that point I like the idea of creating a separate about page as @chuxinh has done on her site).

Thank you for your help again.
